cmake/ExternalProjects.cmake

```diff
@@ -30,6 +30,7 @@ set(PROTOC_EXE /usr/local/bin/protoc)
 set(GRPC_PLUGIN /usr/local/bin/grpc_cpp_plugin)                                  
 
 # Pistache 
+link_directories(${CMAKE_INSTALL_PREFIX}/lib)
```

Comment:
In my local development environment (and for the CI) I needed this to build the tests (otherwise linking to `pistache` failed).

Reply:
Turns out there's the exact thing we need in CMake through the usage of [`IMPORTED`](https://cmake.org/cmake/help/git-stage/guide/importing-exporting/index.html#id5) libraries.

I've also had to tune a bit the `ExternalProject_Add` command. The difference with `spdlog` and `rapidjson` is that they both are header-only libraries hence work just fine with pastint the headers in the right include direction.